analysis/src/macro_expansion.rs

```diff
@@ -0,0 +1,238 @@
+use std::{collections::HashMap, ops::ControlFlow};
+
+use ast::parsed::{
+    asm::{ASMFile, ASMStatement, InstructionBodyElement},
+    postvisit_expression_mut, ArrayExpression, Expression, FunctionDefinition, SelectedExpressions,
+    Statement,
+};
+use number::FieldElement;
+
+pub fn expand<T: FieldElement>(file: ASMFile<T>) -> ASMFile<T> {
+    let mut expander = MacroExpander::default();
+    expander.expand_asm(file)
```

Comment:
Does this macro expander still use the asm/pil context outside of the macro definition?

Reply:
Ah, I guess this is the same as the other file and was just moved to the new crate.

Reply:
Hm it does look different. Can you explain what the differences are?

Reply:
Refactored this based on the current version on `main`, the diff is much smaller now